lib/core/connect.js

```diff
@@ -96,6 +98,8 @@ function buildConnector ({ allowH2, maxCachedSessions, socketPath, timeout, sess
 
       const session = customSession || sessionCache.get(sessionKey) || null
 
+      port = port || 443
```

Comment:
why?

Reply:
I want to see where the socket error is originating.